api/uniter/relation.go

```diff
@@ -48,6 +49,12 @@ func (r *Relation) Life() params.Life {
 	return r.life
 }
 
+// OtherApplication returns the name of the application on the other
+// end of the relation (from this unit's perspective).
+func (r *Relation) OtherApplication() string {
+	return r.otherApp
```

Comment:
It'll be populated with the same app. It's obtained using `relation.RelatedEndpoints(thisApp)`, which uses `counterpartRole` - for peer that returns peer. I think that's correct. It won't have any impact for this change though, since a peer relation couldn't ever be a principal one - the app would have to be both subordinate and insubordinate. :)